bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -858,84 +857,67 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   /**
    * Creates a snapshot from an existing table. NOTE: Cloud Bigtable has a cleanup policy
    *
-   * @param snapshotName a {@link String} object.
+   * @param snapshotId a {@link String} object.
    * @param tableName a {@link TableName} object.
    * @throws IOException if any.
    */
   @Override
-  public void snapshot(String snapshotName, TableName tableName)
+  public void snapshot(String snapshotId, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
 
-    Operation operation = snapshotTable(snapshotName, tableName);
-    try {
-      connection.getSession().getInstanceAdminClient().waitForOperation(operation);
-    } catch (TimeoutException e) {
-      throw new IOException("Timed out waiting for snapshot creation to finish", e);
-    }
+    snapshotTable(snapshotId, tableName);
   }
 
-  /**
-   * Creates a snapshot from an existing table. NOTE: Cloud Bigtable has a cleanup policy
-   *
-   * @param snapshotName a {@link String} object.
-   * @param tableName a {@link TableName} object.
-   * @return a {@link Operation} object.
-   * @throws IOException if any.
-   */
-  protected Operation snapshotTable(String snapshotName, TableName tableName) throws IOException {
-    Preconditions.checkNotNull(snapshotName);
-    Preconditions.checkNotNull(tableName);
-    if (snapshotName.isEmpty()) {
-      // HBase returns an empty operation instance in case snapshotName is an empty string.
-      return Operation.newBuilder().build();
-    }
+  protected Backup snapshotTable(String snapshotId, TableName tableName) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(snapshotId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName.getNameAsString()));
 
-    SnapshotTableRequest.Builder requestBuilder =
-        SnapshotTableRequest.newBuilder()
-            .setCluster(getSnapshotClusterName().toString())
-            .setSnapshotId(snapshotName)
-            .setName(options.getInstanceName().toTableNameStr(tableName.getNameAsString()));
+    CreateBackupRequest request =
+        CreateBackupRequest.of(getBackupClusterName().getClusterId(), snapshotId)
+            .setSourceTableId(tableName.getNameAsString());
 
     int ttlSecs =
-        configuration.getInt(BigtableOptionsFactory.BIGTABLE_SNAPSHOT_DEFAULT_TTL_SECS_KEY, -1);
+        configuration.getInt(
+            BigtableOptionsFactory.BIGTABLE_SNAPSHOT_DEFAULT_TTL_SECS_KEY,
+            BigtableOptionsFactory.BIGTABLE_SNAPSHOT_DEFAULT_TTL_SECS_VALUE);
+
     if (ttlSecs > 0) {
```

Comment:
What happens when ttlSecs < 0?

Reply:
it will throw saying expire time needs to be set. I'll update this